### PR TITLE
Fix: reconcile after configuration's hcl changed

### PR DESCRIPTION
**Pull request by xingming01**
Reconcile again after configuration's hcl changed

Fixes https://github.com/kubevela/terraform-controller/issues/359

### CODE REVIEW

**Review by chivalryq — COMMENTED** [review]
Generally LGTM, please add some test.

**Comment by xingming01** [comment]
> Generally LGTM, please add some test.

Added, thanks

File: controllers/configuration_controller_test.go [hunk 1]
```diff
@@ -2371,10 +2371,36 @@ func TestCheckWhetherConfigurationChanges(t *testing.T) {
...
+			args: args{
+				meta: &TFConfigurationMeta{
+					ConfigurationCMName: "aa",
+					Namespace:           "b",
+					ControllerNamespace: "b",
+				},
+				configurationType: "HCL",
+			},
+			want: want{
+				errMsg: "",
```
Comment (chivalryq) [inline controllers/configuration_controller_test.go:2403]: Case "Configuration is changed" is not tested, which is the point of this PR. Can you add some test for that?

**Review by chivalryq — APPROVED** [review]
LGTM

File: controllers/configuration_controller_test.go (hunk 1 above)
Comment (xingming01) [reply controllers/configuration_controller_test.go:2492]: > Case "Configuration is changed" is not tested, which is the point of this PR. Can you add some test for that?

A case "Configuration changed, and reconcile"" was added

**Pull request merged**